spid_cie_oidc/onboarding/tests/test_03_authn_request.py

```diff
@@ -1,37 +1,47 @@
+import logging
+
 from django.test import TestCase
-from spid_cie_oidc.onboarding.validators.authn_requests import AuthenticationRequestSpid, AuthenticationRequestCie, validate_message
-from . authn_request_settings import AUTHN_REQUEST_SPID, AUTHN_REQUEST_CIE
 from pydantic import ValidationError
-import logging
+from spid_cie_oidc.onboarding.validators.authn_requests import (
+    AuthenticationRequestCie, AuthenticationRequestSpid, validate_message)
+
+from .authn_request_settings import (AUTHN_REQUEST_CIE, AUTHN_REQUEST_SPID,
+                                     AUTHN_REQUEST_SPID_NO_CLIENT_ID,
+                                     AUTHN_REQUEST_SPID_NO_CORRECT_CLAIMS)
 
 logger = logging.getLogger(__name__)
 
 class AuthRequestTest(TestCase):
 
-    def test_authn_1 (self):
+    def validate_spid(self):
+        AuthenticationRequestSpid(**AUTHN_REQUEST_SPID)
+        validate_message(AUTHN_REQUEST_SPID, AuthenticationRequestSpid.get_claims())
 
-        def validate_spid():
-            try:
-                AuthenticationRequestSpid(**AUTHN_REQUEST_SPID)
-                validate_message(AUTHN_REQUEST_SPID, AuthenticationRequestSpid.get_claims())
-                logger.info("ATUHN REQUEST SPID CHECK PASSED")
-            except ValidationError as e:
-                logger.info("CHECK ATUHN REQUEST SPID FAILED, SEE ERRORS:")
-                logger.info(e.json())
-        
-        def validate_cie():
-            try:
-                AuthenticationRequestCie(**AUTHN_REQUEST_CIE)
-                validate_message(AUTHN_REQUEST_CIE, AuthenticationRequestCie.get_claims())
-                logger.info("ATUHN REQUEST CIE CHECK PASSED")
-            except ValidationError as e:
-                logger.info("CHECK ATUHN REQUEST CIE FAILED, SEE ERRORS:")
-                logger.info(e.json())
-
-        logger.info(AuthenticationRequestSpid.schema_json(indent=2))
-        logger.info(AuthenticationRequestCie.schema_json(indent=2))
-        validate_spid()
-        validate_cie()
+    def validate_spid_no_client_id(self):
+        AuthenticationRequestSpid(**AUTHN_REQUEST_SPID_NO_CLIENT_ID) 
+        validate_message(AUTHN_REQUEST_SPID_NO_CLIENT_ID, AuthenticationRequestSpid.get_claims())
+
+    def validate_spid_no_correct_claims(self):
+        AuthenticationRequestSpid(**AUTHN_REQUEST_SPID_NO_CORRECT_CLAIMS) 
+        validate_message(AUTHN_REQUEST_SPID_NO_CORRECT_CLAIMS, AuthenticationRequestSpid.get_claims())
+    
+    def validate_cie(self):
+        AuthenticationRequestCie(**AUTHN_REQUEST_CIE)
+        validate_message(AUTHN_REQUEST_CIE, AuthenticationRequestCie.get_claims())
+
+# logger.info(AuthenticationRequestSpid.schema_json(indent=2))
```

Comment:
what's the purpose of having this?
I mean that you run this single file in your project, good to me but if needed please use the convention

`if __name__ == '__main__':`

anyway, we may use `./manage.py test spid_cie_oidc.onboarding`

you can also append `-k validate_spid_no_client_i` to select only a test